controllers/component_build_controller.go

```diff
@@ -142,6 +143,26 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 func (r *ComponentBuildReconciler) SubmitNewBuild(ctx context.Context, component appstudiov1alpha1.Component) error {
 	log := r.Log.WithValues("Namespace", component.Namespace, "Application", component.Spec.Application, "Component", component.Name)
 
+	// TODO delete this block which is workaround for delayed sync of pvc
+	workspaceStorage := gitops.GenerateCommonStorage(component, "appstudio")
+	existingPvc := &corev1.PersistentVolumeClaim{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: workspaceStorage.Name, Namespace: workspaceStorage.Namespace}, existingPvc); err != nil {
```

Comment:
Since we have a potential race condition with the delayed sync, we probably want to employ the use of https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollImmediate around the Get and Create here